Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -554,6 +554,16 @@ variable (E : EllipticCurve R)
 def j : R :=
   E.Δ'⁻¹ * E.c₄ ^ 3
 
+/-- A variant of `WeierstrassCurve.j_eq_zero_iff` without assuming the ring being reduced. -/
```

Comment:
I think this is `EllipticCurve.j_eq_zero_iff`?

Reply:
Oh sorry.

Reply:
BTW I do think your idea posted on Zulip of changing `EllipticCurve` into a `WeierstrassCurve` with an extra predicate (`IsElliptic`/`IsSmooth`/`IsNeZeroDisc`? Rather than have it be an extended structure) is fantastic, and would prevent issues like this in the future, but I haven't got around it. Have you made any attempts at this?

Reply:
> Have you made any attempts at this?

Not yet. Maybe I'll try later.